SQL ADDED IN LANG FOLDER (#94)
devstrons/heloworld.xyz: lang/sql.md

SQL became a standard of the American National Standards Institute (ANSI) in 1986, and of the International Organization for Standardization (ISO) in 1987. Since then, the standard has been revised to include a larger set of features. Despite the existence of standards, most SQL code requires at least some changes before being ported to different database systems.

### 🤔 Why is SQL so popular?

There are many reasons for the ubiquity of sql , including :

- **Easy to Learn and Use**

- **Mature and Supportive Community**

- **Efficiency, Reliability, and Speed**

- **First-choice Language**

- **The Flexibility of sql Language**

- **Use of sql in academics**

## 📚 Why to use SQL?

> We got why `sql is so popular` at the moment above, but why should one learn how to implement it? Aside from the ease of use and versatility mentioned above, there are several good reasons to learn sql: 

Reviewer: ```suggestion
We know why SQL is so popular at the moment above, but why should one learn how to implement it? Aside from the ease of use and versatility mentioned above, there are several good reasons to learn SQL: 
```

> - **sql developers are in demand :** Across a wide range of fields, there is a [demand for those with sql skills]. If you’re looking to start or change your career, it could be a vital skill to help you. 

Reviewer: ```suggestion
- **SQL developers are in demand:** Across a wide range of fields, there is a demand for those with SQL skills. If you’re looking to start or change your career, it could be a vital skill to help you. 
```

> - **It could lead to a well-paid career :** Data suggests that the median annual salary for those with sql skills is [around $85,000 in the Us].

Reviewer: ```suggestion
- **It could lead to a well-paid career:** Data suggests that the median annual salary for those with SQL skills is around $85,000 in the US.
```

### 🧐 What is sql used for?

> 1. **back-end developement**

Reviewer: ```suggestion
1. **Back-end development**
```

2. **Data analytics**
3. **Finance**

> 4. **marketing**
> 5. **data science**
> 

Reviewer: ```suggestion
4. **Marketing**
5. **Data science**
```

## 🛠️ Tools You Will Need

> You are probably familiar with Windows as they are widely used throughout the world. The latest version of Windows is Windows 10. Generally, you should be on the latest version of the OS, but if you have an older machine or other software requires it, you might be stuck with Windows 7 or older. 

Reviewer: ```suggestion
You are probably familiar with Windows as they are widely used throughout the world. The latest version of Windows is Windows 11. Generally, you should be on the latest version of the OS, but if you have an older machine or other software that requires it, you might be stuck with Windows 7 or older. 
```

> If you go for Windows, there are some shell options for you. Windows comes with the Command line, but also Powershell. Powershell feels more like a normal terminal you would know from the Linux- or Mac world.

Reviewer: ```suggestion
If you go for Windows, there are some shell options for you. Windows comes with the Command line, but also Powershell. Powershell feels more like a normal terminal you would know from the Linux or Mac world.
```

Most text editors and IDEs work on macOS.

> macOS, like Windows, is not Open source, but it comes with the machine. macOS only works on a Mac (officially) so you are tied down to Apple hardware.

Reviewer: ```suggestion
macOS, like Windows, is not Open source, but it comes with the machine. macOS only works on a Mac (officially), so you are tied down to Apple hardware.
```

## ⚙️ Setting up Python development environment

One of the most important things you’ll do when working with any programming language is setup a development environment which allows you to execute the code you write. Without this, you will never be able to check your work and see if your website or application is free of syntax errors.

> #### **⚙️ How to Get Started with SQL**

Reviewer: ```suggestion
### **⚙️ How to Get Started with SQL**
```

**Setting Up Work Environment for Practicing SQL**

If you have no DBMS (Database Management System) already installed on your system, you have several options to choose from. You can install a free, open-source DBMS.

MySQL is the most popular and widely supported open-source database management system. It is very easy to download and use and available for both Windows and Linux (or UNIX) operating system. You can download it freely from here https://dev.mysql.com/downloads/mysql/

You can also install SQL Server Express. It is a free version of Microsoft SQL Server which allows up to 10GB of database storage. You can download the express edition from here https://www.microsoft.com/en-in/download/details.aspx?id=30438.

Alternatively, if you're planning to develop an application with PHP and MySQL you can install WampServer or XAMPP. WampServer is a Windows web development environment. It allows you to create web applications with Apache2, PHP and a MySQL database. It will also provide the MySQL administrative tool PhpMyAdmin to easily manage your databases using a web browser. You can download it from here http://www.wampserver.com/en/

> **⚙️ Which Version Should I Use?**

Reviewer: ```suggestion
### **⚙️ Which Version Should I Use?**
```

Long story short, it’s entirely up to you which version of SQL you use! There’s no right or wrong answer, and it’s painless to make the jump from one version to another should you realize you need to do so later on.

## 💻 Popular SQL IDEs/Editors

- **Adminer**
- **DBeaver**
- **HeidiSQL**
- **RazorSQL**
- **Valentina Studio**

## 🎉 Some cool projects in SQL

- **. Online Phone Shop Display**
- **e-ticket Booking**
- **Book Store Inventory Management**
- **Library Management System**

> - **Bank Database Display** 

Reviewer: Might want to provide links to projects that have been made using SQL. Let us know if you need any help. 😄
